crates/omnix-health/src/check/flake_enabled.rs

```diff
@@ -15,22 +15,156 @@ impl Checkable for FlakeEnabled {
+                val.contains(&"flakes".to_string()) && val.contains(&"nix-command".to_string())
+            }
+        };
+
+        let (info_msg, result) = match nix_info.nix_version.installation_type {
+            NixInstallationType::DeterminateSystems => (
+                "Flakes enabled via Determinate Systems Nix".to_string(),
+                CheckResult::Green,
+            ),
+            NixInstallationType::Official => {
+                if flakes_enabled {
+                    (
+                        format!("experimental-features = {}", val.join(" ")),
+                        CheckResult::Green,
+                    )
+                } else {
+                    (
+                        format!("experimental-features = {}", val.join(" ")),
+                        CheckResult::Red {
+                            msg: "Nix flakes are not enabled".into(),
+                            suggestion: "See https://nixos.wiki/wiki/Flakes#Enable_flakes".into(),
+                        },
+                    )
+                }
+            }
+        };
+
         let check = Check {
             title: "Flakes Enabled".to_string(),
-            info: format!("experimental-features = {}", val.join(" ")),
-            result: if val.contains(&"flakes".to_string())
-                && val.contains(&"nix-command".to_string())
-            {
-                CheckResult::Green
-            } else {
-                CheckResult::Red {
-                    msg: "Nix flakes are not enabled".into(),
-                    suggestion: "See https://nixos.wiki/wiki/Flakes#Enable_flakes".into(),
-                }
-            },
+            info: info_msg,
+            result,
             required: true,
         };
 
         vec![("flake-enabled", check)]
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::traits::CheckResult;
+    use nix_rs::{
+        config::{ConfigVal, NixConfig, TrustedUserValue},
+        flake::system::{Arch, System},
+        info::NixInfo,
+        version::NixVersion,
+    };
+    use url::Url;
+
+    fn mock_config_val<T: Clone>(value: T) -> ConfigVal<T> {
+        ConfigVal {
+            value: value.clone(),
+            default_value: value,
+            description: "Mock config value".to_string(),
+        }
+    }
+
+    fn mock_nix_config_without_flakes() -> NixConfig {
+        NixConfig {
+            cores: mock_config_val(4),
+            experimental_features: mock_config_val(vec!["auto-allocate-uids".to_string()]),
+            extra_platforms: mock_config_val(vec![]),
+            flake_registry: mock_config_val("https://flake-registry.example.com".to_string()),
+            max_jobs: mock_config_val(8),
+            substituters: mock_config_val(vec![Url::parse("https://cache.nixos.org/").unwrap()]),
+            system: mock_config_val(System::Linux(Arch::X86_64)),
+            trusted_users: mock_config_val(vec![TrustedUserValue::User("root".to_string())]),
```

Comment:
Why do you need all these irrelevant settings in the mock config? KISS and as small a code as possible.